sklearn/metrics/ranking.py

```diff
@@ -255,9 +270,38 @@ def _binary_roc_auc_score(y_true, y_score, sample_weight=None):
                                         sample_weight=sample_weight)
         return auc(fpr, tpr, reorder=True)
 
-    return _average_binary_score(
-        _binary_roc_auc_score, y_true, y_score, average,
-        sample_weight=sample_weight)
+    if type_of_target(y_true) != "multiclass":
```

Comment:
We also need to handle the multiclass case in which `type_of_taget(y_true) == 'binary'` but `y_score.shape[1] > 2`